compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1945,15 +1945,15 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
           val resultTpt =
             untpd.InLambdaTypeTree(isResult = true, (tsyms, vsyms) =>
               mt.resultType.substParams(mt, vsyms.map(_.termRef)).substParams(poly, tsyms.map(_.typeRef)))
-          val desugared = desugar.makeClosure(tparams, inferredVParams, body, resultTpt, tree.span)
+          val desugared @ Block(List(defdef), _) = desugar.makeClosure(tparams, inferredVParams, body, resultTpt, tree.span)
```

Comment:
The binding `Block(List(defdef), _)`  (and in next case as well) is not used anywhere. Do you want to ensure the list has exactly one element?

Reply:
That should be a guarantee, but the check isn't necessary anymore, removed it. Thanks